clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -298,6 +298,22 @@ class CoverageMappingBuilder {
                            : SM.getIncludeLoc(SM.getFileID(Loc));
   }
 
+  /// Find out where the current file is included or macro is expanded. If the
+  /// found expansion is a <scratch space>, keep looking.
+  SourceLocation getIncludeOrNonScratchExpansionLoc(SourceLocation Loc) {
```

Comment:
This may be integrated to `getIncludeOrExpansionLoc(FoldScratch=true)` as an optional behavior.